packages/block-library/src/button/editor.scss

```diff
@@ -1,10 +1,10 @@
-.wp-block[data-align="center"] .wp-block-button {
+.wp-block[data-align="center"] > .wp-block-button {
```

Comment:
Is it some sort of block nesting issue that was causing this? What's the difference in using direct descendent selector?

(My worry is that this "fixes" by just... never matching anything)

Reply:
yes, the alignment of the "buttons" block was being applied to the "button" block.

Reply:
The same issue could have happened if you rendered the buttons block inside a "centered" group (not sure that's possible in the UI)